umbral/fragments.py

```diff
@@ -225,7 +227,61 @@ def to_bytes(self):
 
         return serialized_cfrag
 
-    def __bytes__(self):
-        return self.to_bytes()
+    def verify_correctness(self, capsule: "Capsule",
+                       pubkey_a: UmbralPublicKey, pubkey_b: UmbralPublicKey,
+                       params: UmbralParameters=None):
+
```

Comment:
I must say that I prefer to keep the logic in `umbral.pre` and to have this only as a wrapper (didn't we agree on Discord to wrap this?). Now that I think about it, it's the same for `KFrag.verify`.

Reply:
Yeah, I think I prefer this too. Just to keep the logic where the rest of the cryptographic code is.

@cygnusv Am I right that all of the logic described in the Umbral paper is in `umbral.pre`?

Reply:
Ahh, right, we did indeed agree to wrap this - that slipped my mind.

As for keeping all the "cryptographic" code together: it's far from obvious how to do that.  There's code reasonable understood as "cryptography" in every single module, including this one.  

But as for wrapping this, I'll mark this [WIP] and get a fix in later today.  :-)

Reply:
Well, I was talking about Umbral KEM itself. Of course there's cryptographic code in `umbral.point` or `umbral.openssl`, but that's a different abstraction layer, in particular, a lower level. If we move the logic of `CFrag.verify_correctness` and `KFrag.verify` to `umbral.pre`, then I think all the core cryptographic logic of Umbral is in `umbral.pre`, in approximately 600 lines of code. That's neat, IMHO.